lms/djangoapps/verify_student/signals.py

```diff
@@ -1,83 +1,112 @@
 """
-Signal handler for setting default course verification dates
+Signal definitions and functions to send those signals for the verify_student application.
 """
-import logging
 
-from django.core.exceptions import ObjectDoesNotExist
-from django.db.models.signals import post_save
 from django.dispatch import Signal
-from django.dispatch.dispatcher import receiver
-from xmodule.modulestore.django import SignalHandler, modulestore
-
-from common.djangoapps.student.models_api import get_name, get_pending_name_change
-from openedx.core.djangoapps.user_api.accounts.signals import USER_RETIRE_LMS_CRITICAL, USER_RETIRE_LMS_MISC
-
-from .models import SoftwareSecurePhotoVerification, VerificationDeadline, VerificationAttempt
-
-log = logging.getLogger(__name__)
 
+from openedx_events.learning.data import UserData, UserPersonalData, VerificationAttemptData
+from openedx_events.learning.signals import (
+    IDV_ATTEMPT_CREATED,
+    IDV_ATTEMPT_PENDING,
+    IDV_ATTEMPT_APPROVED,
+    IDV_ATTEMPT_DENIED,
+)
 
 # Signal for emitting IDV submission and review updates
 # providing_args = ["attempt_id", "user_id", "status", "full_name", "profile_name"]
 idv_update_signal = Signal()
 
 
-@receiver(SignalHandler.course_published)
-def _listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
+def _create_user_data(user):
     """
-    Catches the signal that a course has been published in Studio and
-    sets the verification deadline date to a default.
+    Helper function to create a UserData object.
     """
-    course = modulestore().get_course(course_key)
-    if course:
-        try:
-            deadline = VerificationDeadline.objects.get(course_key=course_key)
-            if not deadline.deadline_is_explicit and deadline.deadline != course.end:
-                VerificationDeadline.set_deadline(course_key, course.end)
-        except ObjectDoesNotExist:
-            VerificationDeadline.set_deadline(course_key, course.end)
+    user_data = UserData(
+        # NOTE to self: the id field was previously = lms_user_id in edx-exams, which I copied this from
```

Comment:
`User.id` is fine. `lms_user_id` was specific to the `edx-exams` IDA because `User.id` in that application referred to the `id` of the `User` in the `edx-exams` database and not the `id` of the `User` in the `edx-platform` database (which was the purpose of `lms_user_id`).